src/core/qgsmaprenderertask.cpp

```diff
@@ -74,62 +158,53 @@ void QgsMapRendererTask::cancel()
 
 bool QgsMapRendererTask::run()
 {
-  QImage img;
-  std::unique_ptr< QPainter > tempPainter;
-  QPainter *destPainter = mPainter;
-
-#ifndef QT_NO_PRINTER
-  std::unique_ptr< QPrinter > printer;
-#endif // ! QT_NO_PRINTER
+  if ( mErrored )
+    return false;
 
-  if ( mFileFormat == QStringLiteral( "PDF" ) )
+  if ( mGeoPDF )
```

Comment:
Hmm, we need to still keep the option of having a good old geoferenced PDF (vs. a GeoPDF spec PDF). Isn't this killing simple georefenred PDFs?

Reply:
I agree that the georeference pdf needs to be preserved. Allowing someone to have a georeferenced pdf without giving any more information is important when sharing some documents. Since you said that attributes can be shared, sometimes those attributes contains confidential information that should not be shared.

Reply:
Actually that hasn't changed -- it's just been moved as part of the process of making the class thread safe. If geoPDF export ISN'T selected, then export to PDF follows exactly the same logic as currently (it's just moved to different places).

Reply:
@nyalldawson , okay, glad I was mislead by the commit changes ;)